apps/dsb-client-gateway-api/src/app/modules/topic/service/topic.service.ts

```diff
@@ -149,4 +150,34 @@ export class TopicService {
       );
     }
   }
+
+  public async getTopicHistoryByIdAndVersion(id: string, versionNumber: string): Promise<PostTopicDto> {
+
+    const topic: TopicEntity = await this.wrapper.topicRepository.findOne({
+      where: { id, version: versionNumber }
+    });
+
+    return {
+      id: topic.id,
+      name: topic.name,
+      schemaType: topic.schemaType as SchemaType,
+      schema: JSON.stringify(topic.schema),
+      version: topic.version,
+      owner: topic.owner,
+      tags: topic.tags
+    };
+  }
+
+  public async getTopicHistoryById(id: string): Promise<PaginatedTopicResponse> {
+    const [topics, allCount] = await this.wrapper.topicRepository.findAndCount({
+      where: { id }
+    });
+
+    return {
+      count: allCount,
+      limit: 0,
+      page: 1,
+      records: topics as unknown as GetTopicSearchDto[]
```

Comment:
why it's casted? it will return schema and cause topics definition to be invalid

Reply:
from topic entity to GetTopicSearchDto

Reply:
I see what code does but it's incorrect - you should have map function there to transform topic entity to `GetTopicSearchDto`